### PR TITLE
Wrapping plugin block in a check for global window object

### PR DESCRIPTION
**Pull request by nbryant**
- window might not be available, i.e. when rendering React components server-side

### CODE REVIEW

**Comment by yowainwright** [comment]
@nbryant Looks good. 💪  I'll run tests on this, prolly tomorrow. 🔧

~THANK YOU!!!

File: src/shave.js [hunk 1]
```diff
@@ -55,11 +55,13 @@ export default function shave(target, maxHeight, opts) {
...
 }
 
-const plugin = window.$ || window.jQuery || window.Zepto;
-if (plugin) {
-  plugin.fn.extend({
-    shave: function shaveFunc(maxHeight, opts) {
-      return shave(this, maxHeight, opts);
-    },
-  });
+if (typeof window !== 'undefined') {
```
Comment (yowainwright) [inline src/shave.js]: @nbryant first, thanks for the well thought out PR! The only thing I think we might be able to improve is the nested if check. 

I thought of this:
```
const plugin = typeof window !== 'undefined' ? window.$ || window.jQuery || window.Zepto : false;
```
So there would be only 1 if check ...

But the line is pretty long & sometime ternaries look confusing to people (especially when the line is long). 

What are your thoughts?

Comment (nbryant) [inline src/shave.js]: @yowainwright I think you're right on: condensing the expression into one line could save a couple bytes, at the expense of making the code slightly harder to read. One tiny advantage you get from the nested version is that the inner `if (plugin)` expression can get skipped completely if there is no `window` object, which could make it a hair faster. It actually looks like there's actually only 1 byte difference in the size of the minified file, so we're really just splitting hairs - I'll leave it up to you. Thanks for looking at this so quickly!

**Comment by yowainwright** [comment]
Added. Thanks again Nathan! 🍻

**Pull request merged**